code/modules/antagonists/changeling/powers/summon_spiders.dm

```diff
@@ -1,18 +1,191 @@
+		is_operating = FALSE
+		return TRUE
+	is_operating = FALSE
+	return FALSE
+
+/// Child of giant_spider because this should do everything the spider does and more
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider
+	/// References to the owner changeling
+	var/mob/owner_UID
+	/// To check if the spider died via gib or if it still needs to be gibbed, to prevent post-death revival skipping the limit
+	var/gibbed = FALSE
+	/// Handles the spider's behavior
+	var/current_order = IDLE_AGGRESSIVE
+	var/list/enemies = list()
+	sentience_type = SENTIENCE_OTHER
+	venom_per_bite = 3
+	speak_chance = 0
+	wander = 0
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/gib()
+	gibbed = TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/death()
+	if(!gibbed)
+		gib()
+		return
+	var/mob/owner_mob = locateUID(owner_UID)
+	var/datum/action/changeling/spiders/S = locate() in owner_mob.actions
+	if(!isnull(S))
+		S.spider_counter--
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/examine(mob/user)
+	. = ..()
+	if(user.UID() != owner_UID)
+		return
+	switch(current_order)
+		if(IDLE_AGGRESSIVE)
+			. += "<span class='notice'>The giant spider will remain idle but will attack anyone on sight.</span>"
+		if(FOLLOW_AGGRESSIVE)
+			. += "<span class='notice'>The giant spider is following us, but will attack anyone on sight.</span>"
+		if(FOLLOW_RETALIATE)
+			. += "<span class='notice'>The giant spider is following us and staying calm, only attacking if it is attacked.</span>"
+		if(IDLE_RETALIATE)
+			. += "<span class='notice'>The giant spider will remain idle and calm, only attacking if it is attacked.</span>"
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/AltShiftClick(mob/user)
+	. = ..()
+	if(user.UID() != owner_UID)
+		return
+	spider_order(user)
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/spider_order(mob/user)
+	enemies = list()
+	switch(current_order)
+		if(IDLE_AGGRESSIVE)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us but attack anyone on sight.</span>")
+			current_order = FOLLOW_AGGRESSIVE
+			handle_automated_movement()
+		if(FOLLOW_AGGRESSIVE)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us and to remain calm, only attacking if it is attacked.</span>")
+			current_order = FOLLOW_RETALIATE
+			handle_automated_movement()
+		if(FOLLOW_RETALIATE)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle and calm, only attacking if it is attacked.</span>")
+			current_order = IDLE_RETALIATE
+		if(IDLE_RETALIATE)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle, but ready to attack anyone on sight.</span>")
+			current_order = IDLE_AGGRESSIVE
+	handle_automated_movement()
```

Comment:
This is going to call `handle_automated_movement()` twice for some of these calls. You might want to return in some of these cases, unless that was intentional

Reply:
These `handle_automated_movement()` calls are in the Aggressive modes, its been 3 weeks so I don't remember exactly but if I'm not mistaken this being called there made it more smooth when changing modes. Will test it however.

Reply:
Doesn't seem to change much so removed em'